sdk/storage/azure-storage-common/src/crypt.cpp

```diff
@@ -15,91 +15,59 @@
-        for (char c : queryCharacters)
-        {
-          ret[c] = false;
-        }
-        // we also encode % and +
-        ret['%'] = true;
-        ret['+'] = true;
-        // Surprisingly, '=' also needs to be encoded because Azure Storage server side is so
-        // strict. We are applying this function to query key and value respectively, so this won't
-        // affect that = used to separate key and query.
-        ret['='] = true;
-        return ret;
+      const static std::string c_doNotEncodeCharacters = []() {
+        // Core::Http::Url::Encode won't encode unreserved characters.
+        std::string doNotEncodeCharacters = c_subdelimiters;
+        doNotEncodeCharacters += "%/:@?";
+        doNotEncodeCharacters.erase(
+            std::remove_if(
+                doNotEncodeCharacters.begin(),
+                doNotEncodeCharacters.end(),
+                [](char x) {
+                  // we also encode % and +
+                  // Surprisingly, '=' also needs to be encoded because Azure Storage server side is
+                  // so strict. We are applying this function to query key and value respectively,
+                  // so this won't affect that = used to separate key and query.
+                  return x == '%' || x == '+' || x == '=';
+                }),
+            doNotEncodeCharacters.end());
+        return doNotEncodeCharacters;
       }();
-
-      std::string encoded;
-      for (char c : value)
-      {
-        unsigned char uc = c;
-        if (shouldEncodeTable[uc])
-        {
-          encoded += '%';
-          encoded += c_hex[(uc >> 4) & 0x0f];
-          encoded += c_hex[uc & 0x0f];
-        }
-        else
-        {
-          encoded += c;
-        }
-      }
-      return encoded;
+      return Core::Http::Url::Encode(value, c_doNotEncodeCharacters);
     }
 
     std::string UrlEncodePath(const std::string& value)
     {
-      const static std::vector<bool> shouldEncodeTable = []() {
-        std::string pathCharacters
-            = std::string(c_unreserved) + std::string(c_subdelimiters) + "%/:@";
-
-        std::vector<bool> ret(256, true);
-        for (char c : pathCharacters)
-        {
-          ret[c] = false;
-        }
-        // we also encode % and +
-        ret['%'] = true;
-        ret['+'] = true;
-        return ret;
+      const static std::string c_doNotEncodeCharacters = []() {
+        // Core::Http::Url::Encode won't encode unreserved characters.
+        std::string doNotEncodeCharacters = c_subdelimiters;
+        doNotEncodeCharacters += "%/:@";
+        doNotEncodeCharacters.erase(
+            std::remove_if(
+                doNotEncodeCharacters.begin(),
+                doNotEncodeCharacters.end(),
+                [](char x) {
+                  // we also encode % and +
+                  return x == '%' || x == '+';
+                }),
+            doNotEncodeCharacters.end());
+        return doNotEncodeCharacters;
```

Comment:
Similarly, here:
```C++
// or something like this to avoid duplication: c_doNotEncodeQueryParameter + "="
string doNotEncodePath = "!$&'()*,;/:@=";
return Core::Http::Url::Encode(value, doNotEncodePath);
```